File: test-full.js
const http = require('http');

function get(path) {
  return new Promise((resolve, reject) => {
    http.get('http://127.0.0.1:56777' + path, res => {
      let data = '';
      res.on('data', c => data += c);
      res.on('end', () => {
        try { resolve({ status: res.statusCode, data: JSON.parse(data) }); }
        catch(e) { reject(e); }
      });
    }).on('error', reject);
  });
}

async function test() {
  console.log('=== Full Integration Test ===\n');

  // 1. Frontend HTTP
  const feOk = await new Promise(resolve => {
    http.get('http://127.0.0.1:46777/', res => resolve(res.statusCode === 200));
  });
  console.log(feOk ? '✅ Frontend HTTP 200' : '❌ Frontend failed');

  // 2. Backend health
  const health = await get('/api/health');
  console.log(health.data.status === 'ok' ? '✅ Backend health OK' : '❌ Backend failed');

  // 3. Listings with keyword
  const kw = await get('/api/listings?keyword=%E7%B2%BE%E8%A3%85');
  console.log(`✅ Keyword "精装": ${kw.data.total} results`);
  console.log(`   (title: ${kw.data.listings[0]?.title?.substring(0, 30)}...)`);

  // 4. Listings with no filters (should have more)
  const all = await get('/api/listings?page_size=10');
  console.log(`✅ No filters: ${all.data.total} results`);

  // 5. Listing detail with merchant info
  const detail = await get('/api/listings/1');
  const l = detail.data.listing;
  console.log(`✅ Detail API fields:`);
  console.log(`   merchant_name: ${l.merchant_name || 'N/A'}`);
  console.log(`   business_license: ${l.business_license || 'N/A'}`);
  console.log(`   total_deals: ${l.total_deals || 'N/A'}`);
  console.log(`   response_rate: ${l.response_rate || 'N/A'}`);
  console.log(`   avg_response_time: ${l.avg_response_time || 'N/A'}`);
  console.log(`   is_approved: ${l.is_approved !== undefined ? l.is_approved : 'N/A'}`);
  console.log(`   fraud_check: ${l.fraud_check ? 'PRESENT' : 'MISSING'}`);
  if (l.fraud_check) {
    console.log(`     score: ${l.fraud_check.score}`);
    console.log(`     is_flagged: ${l.fraud_check.is_flagged}`);
  }

  // 6. Car-specific filters
  const car = await get('/api/listings?category_code=car&car_mileage=1-3%E4%B8%87%E5%85%AC%E9%87%8C&page_size=5');
  console.log(`✅ Car with mileage filter: ${car.data.total} results`);

  // 7. Job-specific filters
  const job = await get('/api/listings?category_code=job&job_salary_min=10000&page_size=5');
  console.log(`✅ Job with salary filter: ${job.data.total} results`);

  // 8. Admin stats
  const auth = await new Promise(resolve => {
    const req = http.request({
      hostname: '127.0.0.1',
      port: 56777,
      path: '/api/auth/login',
      method: 'POST',
      headers: { 'Content-Type': 'application/json' }
    }, res => {
      let data = '';
      res.on('data', c => data += c);
      res.on('end', () => resolve({ headers: res.headers, data: JSON.parse(data) }));
    });
    req.write(JSON.stringify({ phone: '13800000000', password: 'admin123456' }));
    req.end();
  });

  if (auth.data.token) {
    console.log(`✅ Admin login: token obtained`);
    const stats = await new Promise(resolve => {
      http.get({
        hostname: '127.0.0.1',
        port: 56777,
        path: '/api/admin/stats',
        headers: { 'Authorization': 'Bearer ' + auth.data.token }
      }, res => {
        let data = '';
        res.on('data', c => data += c);
        res.on('end', () => resolve(JSON.parse(data)));
      });
    });
    console.log(`✅ Admin stats: ${Object.keys(stats.stats || {}).length} metrics`);
    console.log(`   users: ${stats.stats?.total_users}, listings: ${stats.stats?.total_listings}, reports: ${stats.stats?.pending_reports}`);

    // 9. Reports
    const reports = await new Promise(resolve => {
      http.get({
        hostname: '127.0.0.1',
        port: 56777,
        path: '/api/admin/reports',
        headers: { 'Authorization': 'Bearer ' + auth.data.token }
      }, res => {
        let data = '';
        res.on('data', c => data += c);
        res.on('end', () => resolve(JSON.parse(data)));
      });
    });
    console.log(`✅ Reports API: ${reports.reports?.length || 0} reports`);

    // 10. Merchant leads
    const leads = await new Promise(resolve => {
      http.get({
        hostname: '127.0.0.1',
        port: 56777,
        path: '/api/merchants/leads',
        headers: { 'Authorization': 'Bearer ' + auth.data.token }
      }, res => {
        let data = '';
        res.on('data', c => data += c);
        res.on('end', () => resolve(JSON.parse(data)));
      });
    });
    console.log(`✅ Leads API: ${leads.leads?.length || 0} leads`);
  }

  console.log('\n=== All tests passed! ===');
}

test().catch(e => {
  console.error('❌ Test failed:', e.message);
  process.exit(1);
});

File: test-full.py
import urllib.request
import json

def test(name, url, method='GET', data=None, token=None):
    headers = {'Content-Type': 'application/json'}
    if token:
        headers['Authorization'] = f'Bearer {token}'
    body = json.dumps(data).encode() if data else None
    req = urllib.request.Request(url, data=body, headers=headers, method=method)
    try:
        resp = urllib.request.urlopen(req, timeout=5)
        result = json.loads(resp.read())
        print(f'  ✓ {name}')
        return result
    except urllib.error.HTTPError as e:
        body = e.read().decode()
        print(f'  ✗ {name}: HTTP {e.code} - {body[:100]}')
        return None
    except Exception as e:
        print(f'  ✗ {name}: {e}')
        return None

BASE = 'http://127.0.0.1:50991/api'
print('=== 通过Vite代理验证完整业务链路 ===\n')

print('1. 基础接口')
test('健康检查', f'{BASE}/health')
print()

print('2. 新人用户流程')
result = test('新人登录', f'{BASE}/auth/login', 'POST', {'phone':'13800138001','password':'123456'})
ct = result['token'] if result else None
if ct:
    test('获取新人资料', f'{BASE}/couple/profile', token=ct)
    cp = test('获取预算', f'{BASE}/couple/budget', token=ct)
    test('获取时间轴', f'{BASE}/couple/timeline', token=ct)
    test('获取备婚攻略', f'{BASE}/guides', token=ct)
    svcs = test('获取服务列表', f'{BASE}/services', token=ct)
    test('获取订单', f'{BASE}/orders', token=ct)
print()

print('3. 商家用户流程')
result = test('商家登录', f'{BASE}/auth/login', 'POST', {'phone':'13800138002','password':'123456'})
mt = result['token'] if result else None
if mt:
    test('商家资料', f'{BASE}/merchant/profile', token=mt)
    test('商家工作台', f'{BASE}/merchant/dashboard', token=mt)
    test('商家案例', f'{BASE}/merchant/cases', token=mt)
    test('商家档期', f'{BASE}/merchant/schedules', token=mt)
    test('商家评价', f'{BASE}/merchant/reviews', token=mt)
    test('商家订单', f'{BASE}/orders', token=mt)
print()

print('4. 管理员流程')
result = test('管理员登录', f'{BASE}/auth/login', 'POST', {'phone':'13800138000','password':'123456'})
at = result['token'] if result else None
if at:
    test('平台统计', f'{BASE}/admin/stats', token=at)
    test('商家列表', f'{BASE}/admin/merchants', token=at)
    test('信用分', f'{BASE}/admin/credit/scores', token=at)
    test('趋势', f'{BASE}/admin/trends', token=at)
    test('漏斗', f'{BASE}/admin/funnel', token=at)
print()

print('5. 创建订单测试')
if ct:
    result = test('创建订单', f'{BASE}/orders', 'POST', {
        'merchant_id': 1,
        'service_id': 1,
        'service_name': '测试婚纱摄影套餐',
        'service_type': '婚纱摄影',
        'total_amount': 8888,
        'deposit_amount': 2000,
        'order_date': '2026-10-01'
    }, token=ct)
    if result and result.get('id'):
        oid = result['id']
        print(f'    订单ID: {oid}, 订单号: {result.get("order_no")}')
        
        if mt:
            test('确认订单', f'{BASE}/orders/{oid}/confirm', 'PUT', token=mt)
            test('标记到店', f'{BASE}/orders/{oid}/visit', 'PUT', token=mt)
            test('标记交付', f'{BASE}/orders/{oid}/deliver', 'PUT', token=mt)
            test('评价订单', f'{BASE}/orders/{oid}/review', 'PUT', {
                'rating': 5,
                'content': '非常满意，服务很棒！'
            }, token=ct)

print('\n=== 验证完成 ===')
